### PR TITLE
fix dv.len < num_rows bug

### PR DESCRIPTION
**Pull request by nicklan**
We have a bug where the DV can be shorter than the # of rows returned, and we'll panic because we try and `split_off` with a len > than the dv len.

The added function also allows extending the vector with a specified value in the case the engine wants things to be of equal length, like as is required by arrow's `filter_record_batch`.

This is allowed, and the semantics are just that missing items are included in the query. This adjusts to do that, and notes the correct semantics in the comments.

### CODE REVIEW

**Review by sppalkia — APPROVED** [review]

File: kernel/src/actions/deletion_vector.rs [hunk 1]
```diff
@@ -223,6 +223,19 @@ pub(crate) fn treemap_to_bools(treemap: RoaringTreemap) -> Vec<bool> {
     }
 }
 
+/// helper function to split an `Option<Vec<bool>>`. Because deletion vectors apply to a whole file,
+/// but parquet readers can chunk the file, there is a need to split the vector up. This takes an
+/// optional vector, and if it's `Some`, splits it at `split_index`. The passed vector is modified
+/// in place, and the split off component is returned. If `vector` is `None`, or if the
+/// `split_index` is >= to the length of the vector, the passed vector is left untouched and `None`
+/// is returned.
+pub fn split_vector(vector: Option<&mut Vec<bool>>, split_index: usize) -> Option<Vec<bool>> {
```
Comment (zachschuermann) [inline kernel/src/actions/deletion_vector.rs:232]: nit: do we intend this helper to be pub?

Comment (zachschuermann) [reply kernel/src/actions/deletion_vector.rs:232]: i suppose for those that rely on scan data API they would all have to implement something like this?

**Review by zachschuermann — APPROVED** [review]
lgtm

File: kernel/src/actions/deletion_vector.rs (hunk 1 above)
Comment (nicklan) [reply kernel/src/actions/deletion_vector.rs]: Yep. Our examples want to use it, for example, so it already needs to be pub

**Pull request merged**